spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -227,6 +227,8 @@ public PullRequest createPullRequest(String subscriptionName, Integer maxMessage
 						PubSubSubscriptionUtils.toProjectSubscriptionName(subscriptionName, this.projectId).toString());
 
 		if (maxMessages != null) {
```

Comment:
Side note: I think it is a mistake that maxMessage can be set to `null`, because according to the Google PubSub API docs it must always have a positive number (see: https://cloud.google.com/pubsub/docs/reference/rpc/google.pubsub.v1#pullrequest). I believe that if it isn't set, then a default value of `0` is used.

Shall I report it as a bug?

Reply:
Yes, please. Zero is disallowed, too, so we should assert that the parameter is neither null nor zero.

Reply:
I created the bug #2248